amethyst_utils/src/ortho_camera.rs

```diff
@@ -0,0 +1,103 @@
+use amethyst_core::Axis2;
+use amethyst_core::cgmath::Ortho;
+use amethyst_core::specs::{Component, DenseVecStorage, Join, System, ReadExpect, ReadStorage, WriteStorage};
+use amethyst_renderer::{Camera, ScreenDimensions};
+
+/// `Component` attached to the camera's entity that allows automatically adjusting the camera's matrix according to preferences in the "mode" field.
+#[derive(Default)]
+pub struct NormalOrthoCamera {
+    /// How the camera's matrix is changed when the window's aspect ratio changes. See `CameraNormalizeMode` for more info.
+    pub mode: CameraNormalizeMode,
+}
+
+impl NormalOrthoCamera {
+    pub fn camera_offsets(&self, ratio: f32) -> (f32,f32,f32,f32) {
+        self.mode.camera_offsets(ratio)
+    }
+}
+
+impl Component for NormalOrthoCamera {
+    type Storage = DenseVecStorage<Self>;
+}
+
+/// Settings that decide how to scale the camera's matrix when the aspect ratio changes.
+pub enum CameraNormalizeMode {
+    /// Using an aspect ratio of 1:1, tries to ajust the matrix values of the camera so
+    /// that the direction opposite to the stretch_direction is always [0,1].
+    /// Scene space can be lost on the specified stretch_direction.
```

Comment:
Could you please reformulate a bit? What does it mean to have a direction be always [0,1]?